superset-frontend/src/dashboard/components/FiltersBadge/selectors.ts

```diff
@@ -132,12 +132,12 @@ export const selectIndicatorsForChart = (
   // for now we only need to know which columns are compatible/incompatible,
   // so grab the columns from the applied/rejected filters
   const appliedColumns: Set<string> = new Set(
-    (chart?.queryResponse?.applied_filters || []).map(
+    (chart?.queriesResponse?.[0]?.applied_filters || []).map(
```

Comment:
@etr2460 Hi I see you implemented this part of code, I refactored it to use multi queries, can you please look that my changes not break some functionality. Thanks

Reply:
I implemented this logic. The goal of these two lines is to get a set of all the columns that were applied on a chart, and the columns that were rejected, in order to display them in the filter badge.

If a chart makes multiple queries, it would probably be more correct to add the `applied_filters` and `rejected_filters` from each of the queries, instead of just the first one. But that really depends on how dashboard filters works with multiple queries.

Reply:
> If a chart makes multiple queries, it would probably be more correct to add the `applied_filters` and `rejected_filters` from each of the queries, instead of just the first one. But that really depends on how dashboard filters works with multiple queries.

I think we should probably be applying the same filters to all `QueryObject`s. Since a `QueryContext` currently only supports a single datasource, the applied/rejected columns should be identical. However, temporal filters will only be applicable for `QueryObject`s with `is_temporal: true`.